src/NzbDrone.Core/DecisionEngine/Specifications/RssSync/HistorySpecification.cs

```diff
@@ -116,6 +124,41 @@ public virtual DownloadSpecDecision IsSatisfiedBy(RemoteEpisode subject, SearchC
                             return DownloadSpecDecision.Reject(DownloadRejectionReason.HistoryUpgradesNotAllowed, "{0} grab event in history and Quality Profile '{1}' does not allow upgrades", rejectionSubject, qualityProfile.Name);
                     }
                 }
+
+                if (episode is { HasFile: true })
+                {
+                    EpisodeHistory availableUsableEpisodeHistoryForCustomFormatScore;
+
+                    var episodeHistoryEventTypeForHistoryComparison = new List<EpisodeHistoryEventType>
+                    {
+                        EpisodeHistoryEventType.Grabbed,
+                        EpisodeHistoryEventType.SeriesFolderImported,
+                    };
+
+                    if (episodeHistoryEventTypeForHistoryComparison.Contains(mostRecent.EventType))
+                    {
+                        availableUsableEpisodeHistoryForCustomFormatScore = mostRecent;
+                    }
+                    else
+                    {
+                        availableUsableEpisodeHistoryForCustomFormatScore = _historyService.MostRecentForEpisodeInEventCollection(
+                            episode.Id,
+                            new ReadOnlyCollection<EpisodeHistoryEventType>(episodeHistoryEventTypeForHistoryComparison));
+                    }
+
+                    if (availableUsableEpisodeHistoryForCustomFormatScore == null)
+                    {
+                        continue;
+                    }
+
+                    var mostRecentCustomFormat = _formatService.ParseCustomFormat(availableUsableEpisodeHistoryForCustomFormatScore, subject.Series);
+                    var mostRecentCustomFormatScore = qualityProfile.CalculateCustomFormatScore(mostRecentCustomFormat);
+
+                    if (mostRecentCustomFormatScore > subject.CustomFormatScore)
+                    {
+                        return DownloadSpecDecision.Reject(DownloadRejectionReason.HistoryCustomFormatScore, "Quality Profile '{0}' has a higher Custom Format score than the report: {1}", qualityProfile.Name, subject.CustomFormatScore);
```

Comment:
None of our other messaging indicates the quality profile name and instead indicates which item is being compared, be it file or release in queue, we should maintain that here as well.